libs/libc/stdlib/lib_rand48.c

```diff
@@ -0,0 +1,194 @@
+/****************************************************************************
+ * libs/libc/stdlib/lib_rand48.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <sys/types.h>
+#include <stdlib.h>
+
+#include <nuttx/lib/lib.h>
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static unsigned short g_seed48[7] =
+{
+  0,
+  0,
+  0,
+  0xe66d,
+  0xdeec,
+  0x5,
+  0xb
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static uint64_t rand48_step(FAR unsigned short *xi, FAR unsigned short *lc)
+{
+  uint64_t a, x;
+
+  x = xi[0] | (xi[1] + (0ul << 16)) | (xi[2] + (0ull << 32));
+  a = lc[0] | (lc[1] + (0ul << 16)) | (lc[2] + (0ull << 32));
+  x = a * x + lc[3];
+
+  xi[0] = x;
+  xi[1] = x >> 16;
+  xi[2] = x >> 32;
+  return x & 0xffffffffffffull;
+}
```

Comment:
should this be guarded by `#ifdef CONFIG_HAVE_LONG_LONG`?

Reply:
Sorry for the late update. Done, Thanks!